specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json

```diff
@@ -0,0 +1,704 @@
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ErrorResponse": {
+      "description": "The error response send when an operation fails.",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "error code"
+        },
+        "message": {
+          "type": "string",
+          "description": "error message"
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ]
+    },
+    "Workspace": {
+      "type": "object",
+      "description": "An object that represents a machine learning workspace.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/WorkspaceProperties",
+          "description": "The properties of the machine learning workspace.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "WorkspaceProperties": {
+      "type": "object",
+      "description": "The properties of a machine learning workspace.",
+      "required": [
+        "userStorageAccountId"
+      ],
+      "properties": {
+        "userStorageAccountId": {
+          "description": "The fully qualified arm id of the storage account associated with this workspace.",
+          "type": "string"
+        },
+        "ownerEmail": {
+          "description": "The email id of the owner for this workspace.",
+          "type": "string"
+        },
+        "workspaceType": {
+          "type": "string",
+          "enum": [
+            "Production",
+            "Free",
+            "Anonymous",
+            "PaidStandard",
+            "PaidPremium"
+          ],
+          "x-ms-enum": {
+            "name": "WorkspaceType",
+            "modelAsString": false
+          },
+          "readOnly": true,
+          "description": "The type of this workspace."
+        },
+        "workspaceState": {
+          "type": "string",
+          "enum": [
+            "Deleted",
+            "Enabled",
+            "Disabled",
+            "Migrated",
+            "Updated",
+            "Registered",
+            "Unregistered"
+          ],
+          "x-ms-enum": {
+            "name": "WorkspaceState",
+            "modelAsString": false
+          },
+          "readOnly": true,
+          "description": "The current state of workspace resource."
+        },
+        "workspaceId": {
+          "description": "The immutable id associated with this workspace.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationTime": {
+          "description": "The creation time for this workspace resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "workspaceAuthorizationToken": {
```

Comment:
should not be returned in the response of GET/PUT/PATCH. If needed, this should be exposed through a separate POST API

Reply:
This does not get populated in the GET request. However, we do return it as part of a PUT and PATCH request. Also, we do have a POST API which return this key. Not sure why returning it in a POST/PATCH response is an issue. Can you specify why having this as part of PUT/PATCH response is not ok? Readers on this resource won't see the key as we do not populate this field in the GET response. Contributors/Owners will be able to see this by using a PUT/POST/PATCH operation.

Reply:
secrets should not be returned in PUT. Soon the Activity Logs in Azure will start logging responses from PUT/GET and we don't want secrets in that stream. If the POST API to get the key is there, then that should be the one that is used. Returning it in PUT does not add any benefit

Reply:
Fixed